src/pages/chat/Chat.tsx

```diff
@@ -85,30 +87,36 @@ const Chat = () => {
   }, [selectedChat, fetchData])
 
   useEffect(() => {
-    if (selectedChat && messages.length > 0) {
+    if (selectedChat && messages.length) {
       scrollRef.current?.scrollTo({ top: scrollRef.current.scrollHeight })
     }
   }, [selectedChat, messages.length])
 
-  const messagesList = messages.map((message) => (
-    <Message key={message._id} message={message} />
-  ))
-
   if (loading) {
     return <Loader size={100} />
   }
 
+  const messagesListWithDate = groupedMessages.map((group) => (
+    <Box key={group.date} sx={styles.messagesWithDate}>
+      <ChatDate date={group.date} />
+      {group.messages.map((message, index) => (
+        <Message
+          key={message._id}
+          message={message}
+          prevMessage={index ? group.messages[index - 1] : null}
```

Comment:
Maybe better use index && group.messages[index - 1]

Reply:
As I remember in that case prevMessage will be message or false